code/game/machinery/computer/shuttle_computers.dm

```diff
@@ -426,6 +455,20 @@
 		to_chat(usr, "<span class='info'>You insert \the [SC] into \the [src].</span>")
 		src.updateUsrDialog()
 
+/obj/machinery/computer/shuttle_control/proc/use_pass(obj/item/weapon/card/shuttle_pass/P, mob/user)
+	if(!istype(P))
+		return
+
+	if(user.drop_item(P, src))
+		if(shuttle && shuttle.type == P.allowed_shuttle)
+			if(shuttle.travel_to(P.destination, src, user))
```

Comment:
But what happens if the condition fails? The user has already inserted the card, and then it just stays in the machine forever.